src/exo/LoopIR_scheduling.py

```diff
@@ -1257,28 +1279,14 @@ def map_s(self, s):
 
         if isinstance(s, (LoopIR.Assign, LoopIR.Reduce)):
             e = self.exprs[0]
-            # bind LHS when self.use_cse == True
-            if (
-                self.use_cse
-                and isinstance(e, LoopIR.Read)
-                and e.name == s.name
-                and e.type == s.type
-            ):
-                try:
-                    for i, j in zip(e.idx, s.idx):
-                        Check_ExprEqvInContext(self.orig_proc, [s], i, j)
-
-                    return [
-                        s.update(
-                            name=self.new_name,
-                            cast=None,
-                            idx=[],
-                            rhs=self.apply_e(s.rhs),
-                            eff=None,
-                        )
-                    ]
-                except SchedulingError:
-                    pass
+            if (new_rhs := self.map_e(s.rhs)) is None:
+                new_rhs = s.rhs
```

Comment:
Better would be to return `None` when `new_rhs` is `None` because otherwise the `return [s.update(rhs=new_rhs)]` makes a pointless copy of `s`. Naturally, don't skip the `self.found_write` logic when making this change.